mmv1/products/dataplex/Datascan.yaml

```diff
@@ -0,0 +1,640 @@
+          - !ruby/object:Api::Type::NestedObject
+            name: 'schedule'
+            description: |
+             The scan is scheduled to run periodically.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'cron'
+                required: true
+                description: Cron schedule for running scans periodically. This field is required for Schedule scans.
+      - !ruby/object:Api::Type::String
+        name: 'field'
+        immutable: true
+        description: |
+          The unnested field (of type Date or Timestamp) that contains values which monotonically increase over time. If not specified, a data scan will run for all data in the table.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'executionStatus'
+    output: true
+    description: |
+      Status of the data scan execution.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'latestJobEndTime'
+        description: |
+          The time when the latest DataScanJob started.
+      - !ruby/object:Api::Type::String
+        name: 'latestJobStartTime'
+        description: |
+          The time when the latest DataScanJob ended.
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
+    output: true
+    description: |
+      The type of DataScan.
+    values:
+      - :DATA_SCAN_TYPE_UNSPECIFIED
+      - :DATA_QUALITY
+      - :DATA_PROFILE
+  - !ruby/object:Api::Type::NestedObject
+    name: 'dataQualitySpec'
+    description: |
+      DataQualityScan related setting.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'rules'
+        description: |
+          The list of rules to evaluate against a data source. At least one rule is required.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+             - !ruby/object:Api::Type::String
+               name: 'column'
+               description: |
+                 The unnested column which this rule is evaluated against.
+             - !ruby/object:Api::Type::Boolean
+               name: 'ignoreNull'
+               description: |
+                 Rows with null values will automatically fail a rule, unless ignoreNull is true. In that case, such null rows are trivially considered passing. Only applicable to ColumnMap rules.
+             - !ruby/object:Api::Type::String
+               name: 'dimension'
+               required: true
+               description: |
+                 The dimension a rule belongs to. Results are also aggregated at the dimension level. Supported dimensions are ["COMPLETENESS", "ACCURACY", "CONSISTENCY", "VALIDITY", "UNIQUENESS", "INTEGRITY"]
+             - !ruby/object:Api::Type::Integer
+               name: 'threshold'
+               description: |
+                The minimum ratio of passing_rows / total_rows required to pass this rule, with a range of [0.0, 1.0]. 0 indicates default value (i.e. 1.0).
+               default_value: 0
```

Comment:
i might be reading this wrong, but it seems to me that if the user inputs `0` that the default value (`1.0`) will be used. have you tested this to see how this works by chance to see if entering `0` returns `0`? just curious

Reply:
(will double check this and update here)

The API behavior is that specifying 0, or not specifying a value at all, both cause the system to interpret the threshold as 1.0.


The API Read behavior however should return whatever the user specified --- i.e. 0 / unspecified.